fdbserver/workloads/ConsistencyCheck.actor.cpp

```diff
@@ -1933,6 +1938,89 @@ struct ConsistencyCheckWorkload : TestWorkload {
 		return true;
 	}
 
+	// Checks if the blob workers are "correct".
+	// Returns false if ANY of the following
+	// - any blob worker is on a diff DC than the blob manager/CC, or
+	// - any worker that should have a blob worker does not have one, or
+	// - any worker that should NOT have a blob worker does indeed have one
+	ACTOR Future<bool> checkBlobWorkers(Database cx,
+	                                    DatabaseConfiguration configuration,
+	                                    ConsistencyCheckWorkload* self) {
+		state vector<BlobWorkerInterface> blobWorkers = wait(getBlobWorkers(cx));
+		state vector<WorkerDetails> workers = wait(getWorkers(self->dbInfo));
+		state std::set<NetworkAddress> blobWorkerAddrs;
+		Optional<Key> ccDcId;
+		NetworkAddress ccAddr = self->dbInfo->get().clusterInterface.clientInterface.address();
+
+		// get the CC's DCID
+		for (const auto& worker : workers) {
+			if (ccAddr == worker.interf.address()) {
+				ccDcId = worker.interf.locality.dcId();
+			}
+		}
+
+		if (!ccDcId.present()) {
+			TraceEvent("ConsistencyCheck_DidNotFindCC");
+			return false;
+		}
+
+		for (const auto& bwi : blobWorkers) {
+			if (bwi.locality.dcId() != ccDcId) {
+				TraceEvent("ConsistencyCheck_BWOnDiffDcThanCC")
+				    .detail("BWID", bwi.id())
+				    .detail("BwDcId", bwi.locality.dcId())
+				    .detail("CcDcId", ccDcId);
+				return false;
+			}
+			blobWorkerAddrs.insert(bwi.stableAddress());
+		}
+
+		for (int i = 0; i < workers.size(); i++) {
+			NetworkAddress addr = workers[i].interf.stableAddress();
+			if (!configuration.isExcludedServer(workers[i].interf.addresses())) {
+				if (workers[i].processClass == ProcessClass::BlobWorkerClass) {
+					// this is a worker with processClass == BWClass, so should have a blob worker
+					if (blobWorkerAddrs.find(addr) == blobWorkerAddrs.end()) {
+						TraceEvent("ConsistencyCheck_NoBWOnBWClass").detail("Address", addr);
+						return false;
+					}
+
+					// FIXME: improve this check (relies on simulation). It's currently brittle since it relies on
```

Comment:
TODO: discuss with Josh

Reply:
I think it might be sufficient to have the other roles fitness checks ensure that they're not placed on the blob worker class.
After looking at consistency check more that's how it works for everything else.

Reply:
Yeah I think fitness checks are the way to go, especially since it's not entirely true that other roles will never be recruited onto a blob worker process class (e.g. SS doing critical recruitment).